crates/predicate/src/key.rs

```diff
@@ -0,0 +1,149 @@
+//! Predicate key implementation and type registry.
+
+use crate::constants::{NEVER_ACCEPT_PREDICATE_TYPE, PredicateType};
+use crate::errors::Result;
+
+/// A predicate key encodes a formal boolean statement with a type identifier.
+///
+/// As defined in the SPS-predicate-fmt specification, a predicate key consists of:
+/// - `type`: u8 - Indicates which predicate backend to use
+/// - `condition`: bytes - Backend-specific predicate condition data
+///
+/// Each predicate type has its own interpretation of the condition data:
+///
+/// An empty predicate key represents the "never accept" predicate.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct PredicateKey {
+    /// Raw bytes containing the type (first byte) and condition data (remaining bytes).
+    /// If empty, represents the "never accept" predicate type.
+    data: Vec<u8>,
+}
```

Comment:
Given the docstring I think we should break `data` into `type` and `condition`. And create a `TryFrom` for `&[u8]`.

Reply:
It probably makes sense to have multiple different types for different representations.  One for what you describe when constructing things and another like this for when parsing things zero-copy and retaining the ability to get hashes of them easily:

```rs
pub struct PredicateKeyBuf<'b> {
    buf: &'b [u8]
}
```